go/interpreter/lfvm/instructions_test.go

```diff
@@ -1758,3 +1758,70 @@ func TestInstructions_ComparisonAndShiftOperations(t *testing.T) {
 		})
 	}
 }
+
+func TestInstructions_Sha3_ReportsOutOfGasa(t *testing.T) {
+	tests := map[string]struct {
+		size          uint64
+		expectedError error
+	}{
+		"memory expansion": {
+			size:          64,
+			expectedError: errOutOfGas,
+		},
+		"dynamic gas price": {
+			size:          1,
+			expectedError: errOutOfGas,
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			ctxt := context{gas: 3}
+			ctxt.memory = NewMemory()
+			ctxt.stack = NewStack()
+			ctxt.stack.push(uint256.NewInt(test.size))
+			ctxt.stack.push(uint256.NewInt(0))
+			err := opSha3(&ctxt)
+			if err != test.expectedError {
+				t.Fatalf("unexpected error, wanted %v, got %v", test.expectedError, err)
+			}
+		})
+	}
+}
+
+func TestInstructions_Sha3_WritesCorrectHashInStack(t *testing.T) {
+
+	want := tosca.Hash{
+		0xbc, 0x36, 0x78, 0x9e, 0x7a, 0x1e, 0x28, 0x14,
+		0x36, 0x46, 0x42, 0x29, 0x82, 0x8f, 0x81, 0x7d,
+		0x66, 0x12, 0xf7, 0xb4, 0x77, 0xd6, 0x65, 0x91,
+		0xff, 0x96, 0xa9, 0xe0, 0x64, 0xbc, 0xc9, 0x8a}
+
+	ctxt := getEmptyContext()
```

Comment:
please rewrite this test with 2 cases.
We just dont have this kind of arrangement in the rest of the file and it is easier to find what we are testing if we follow a declarative approach.